src/components/graph/graph-types.ts

```diff
@@ -0,0 +1,29 @@
+export type uuid = string
+export type slug = string
+
+export type nodeData = {
```

Comment:
@SleeplessByte We're currently using `type` and `interface` in various places. When should we be using which?

Reply:
Yes, I don't know the answer to that either, it would be great to learn!

Reply:
See: https://microsoft.github.io/TypeScript-New-Handbook/everything/#interface-vs-alias

Basically: it doesn't matter, but here are my rules:

- ComponentProps: use `interface` most of the time
- Actual interfaces (things that you want to `implement`), use `interface`
- Reducer state: `interface`
- Reducer action: `type`
- Anything that is union: `type`
- Actual aliases (such as `Slug`): `type`.

Just ask me if you don't know which one to use, but TS will tell you when ya fucked up.

So in this document, all the `Data` types, i'd make interfaces, because you'll use them as such:

```
const myObject: ThatDataInterface = {
  ...
}
```

Reply:
Some people have different rules because interface declarations merge (so you can literally extend a type outside a library) and types don't, but whatev. That doesn't matter here, because we're not building a library.

Reply:
> but TS will tell you when ya fucked up

It sure will... :sweat_smile: